File: src/Lesson70/JAVA/controller/GameController.java
package Lesson70.JAVA.controller;

import Lesson70.JAVA.model.Board;
import Lesson70.JAVA.model.Player;
import Lesson70.JAVA.model.Symbol;

import java.util.Scanner;

public class GameController {
    Player player1;
    Player player2;

    Board board;

    {
        initPlayers();
    }

    private void initPlayers() {
        Scanner in = new Scanner(System.in);
        System.out.println("Enter the name of Player 1");
        String playerOne = in.nextLine();

        do {
            System.out.println("Choose a symbol for Player " + playerOne + " (X or O): ");
            String symbol = in.nextLine();
            if (symbol.length() == 1) {
                if (symbol.toUpperCase().equals("X")) {
                    player1 = new Player(playerOne, Symbol.X);
                } else {
                    player1 = new Player(playerOne, Symbol.O);
                }
            }
        } while (player1 == null);

        System.out.println("Enter the name of Player 2");
        String playerTwo = in.nextLine();
        if (Symbol.X == player1.getSymbol()) {
            player2 = new Player(playerTwo, Symbol.O);
        } else {
            player2 = new Player(playerTwo, Symbol.X);
        }
        //in.close();
    }

    public GameController() {
        board = new Board();
    }

    public void playGame() {
        Scanner in = new Scanner(System.in);
        Player currentPlayer = player1;
        while (true) {
            board.showCurrentState();
            System.out.println("Player " + currentPlayer.getName() + " enter your movie(row and column): ");
            int row = 0, column = 0;
            while ((row < 1 || row > 3) && (column < 1 || column > 3)) {
                row = in.nextInt();
                column = in.nextInt();
            }
            row--;
            column--;

            if (board.isCellEmpty(row, column)) {
                board.setCell(row, column, currentPlayer.getSymbol());
                board.showCurrentState();
            }

            if (board.isWinner(currentPlayer)) {
                System.out.println("The player " + currentPlayer.getName() + " is win!!! GAME OVER!!!");
                break;
            } else {
                currentPlayer = (currentPlayer == player1 ? player2 : player1);
            }
        }
    }
}

File: src/Lesson70/Kotlin/controller/GameController.kt
package Lesson70.Kotlin.controller

import Lesson70.Kotlin.model.Board
import Lesson70.Kotlin.model.Player
import Lesson70.Kotlin.model.Symbol

class GameController {
    private lateinit var player1: Player
    private lateinit var player2: Player

    private val board = Board()

    init {
        initPlayer()
    }

    fun playGame() {
        var currentPlayer = player1

        while (true) {
            board.showCurrentState()
            println("Ход игрока ${currentPlayer.name} (${currentPlayer.symbol})")
            println("Введите строку и столбец от 1 до 3 через пробел:")

            var row: Int
            var column: Int

            while (true) {
                val input = readln().trim().split(" ")

                if (input.size != 2) {
                    println("Нужно ввести 2 числа, например: 2 3")
                    continue
                }

                row = input[0].toIntOrNull()?.minus(1) ?: -1
                column = input[1].toIntOrNull()?.minus(1) ?: -1

                if (row !in 0..2 || column !in 0..2) {
                    println("Координаты должны быть от 1 до 3. Попробуй снова.")
                    continue
                }

                if (!board.isCellEmpty(row, column)) {
                    println("Клетка уже занята. Выбери другую.")
                    continue
                }

                break
            }

            board.setCell(row, column, currentPlayer.symbol)

            if (board.isWinner(currentPlayer.symbol)) {
                board.showCurrentState()
                println("Игрок ${currentPlayer.name} победил! GAME OVER!")
                break
            }

            if (board.isFull()) {
                board.showCurrentState()
                println("Ничья! GAME OVER!")
                break
            }

            currentPlayer = if (currentPlayer == player1) player2 else player1
        }
    }

    private fun initPlayer() {
        println("Введите имя игрока 1:")
        val namePlayer1 = readln()

        println("Выберите символ для игрока 1 (X или O):")
        val symbolFromConsole = readln().trim().uppercase()

        val symbol = when (symbolFromConsole) {
            "X", "Х" -> Symbol.X
            "O", "О" -> Symbol.O
            else -> {
                println("Некорректный ввод. По умолчанию выбран X.")
                Symbol.X
            }
        }

        player1 = Player(namePlayer1, symbol)

        println("Введите имя игрока 2:")
        val namePlayer2 = readln()

        player2 = Player(
            namePlayer2,
            if (player1.symbol == Symbol.X) Symbol.O else Symbol.X
        )
    }
}